### PR TITLE
*: use values to reduce GC pressure 

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve? 

Use prefer using values for channel types to reduce GC pressure. Same as https://github.com/pingcap/ticdc/pull/2415

Left: master
Right: this patch

Max (master): ~80ms
Max (this patch): ~35ms
image
image
image
image

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
None
```

### CODE REVIEW

File: pkg/workerpool/pool_impl.go [hunk 1]
```diff
@@ -280,9 +280,6 @@ func (w *worker) run(ctx context.Context) error {
 		case <-ctx.Done():
 			return errors.Trace(ctx.Err())
 		case task := <-w.taskCh:
-			if task == nil {
-				return cerrors.ErrWorkerPoolEmptyTask.GenWithStackByArgs()
-			}
```
Comment (amyangfei) [inline pkg/workerpool/pool_impl.go:285]: Should we check empty task here?

File: pkg/workerpool/pool_impl.go [hunk 2]
```diff
@@ -246,7 +246,7 @@ type task struct {
 }
 
 type worker struct {
-	taskCh       chan *task
+	taskCh       chan task
```
Comment (amyangfei) [inline pkg/workerpool/pool_impl.go:249]: What about adding a benchmark to the workerpool after this change

**Review by amyangfei — COMMENTED** [review]
Curious about whether CPU usage has any change after this memory optimization?

**Comment by overvenus** [comment]
> Curious about whether CPU usage has any change after this memory optimization?

@amyangfei No significant change on CPU usage, see PR description.

**Comment by amyangfei** [comment]
> > Curious about whether CPU usage has any change after this memory optimization?
> 
> @amyangfei No significant change on CPU usage, see PR description.

From the monitor can we draw the following conclusions, after using values

- Average CPU usage increases a little(a **rough** observation), but not significant compared to the whole CPU usage
- CPU usage is more stable, fewer fluctuation in monitor

File: pkg/workerpool/pool_impl.go (hunk 1 above)
Comment (overvenus) [reply pkg/workerpool/pool_impl.go:285]: I don't think it necessary as task is used only in this file, and it never be empty.

**Comment by overvenus** [comment]
```
# Master
goos: linux
goarch: amd64
pkg: github.com/pingcap/ticdc/pkg/workerpool
cpu: Intel(R) Xeon(R) CPU E5-2630 v4 @ 2.20GHz
BenchmarkWorkerpool-40            476822              2585 ns/op             272 B/op          5 allocs/op

# This patch
goos: linux
goarch: amd64
pkg: github.com/pingcap/ticdc/pkg/workerpool
cpu: Intel(R) Xeon(R) CPU E5-2630 v4 @ 2.20GHz
BenchmarkWorkerpool-40            480866              2453 ns/op             256 B/op          4 allocs/op
```

**Review by amyangfei — APPROVED** [review]

**Review by ben1009 — APPROVED** [review]

**Comment by overvenus** [comment]
#2201 reduces 3 allocations, well done @liuzix !

```
goos: linux
goarch: amd64
pkg: github.com/pingcap/ticdc/pkg/workerpool
cpu: Intel(R) Xeon(R) CPU E5-2630 v4 @ 2.20GHz
BenchmarkWorkerpool-40            837417              1579 ns/op              56 B/op          1 allocs/op
PASS
ok      github.com/pingcap/ticdc/pkg/workerpool 2.343s
```

**Pull request merged**